File: Golang/Merge-k-Sorted-Lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeKLists(lists []*ListNode) *ListNode {
    t := []int{}
    
    for _, list := range lists {
        current := list
        for current != nil {
            t = append(t, current.Val)
            current = current.Next
        }
    }
    
    sort.Slice(t, func(i, j int) bool {
        return t[i]<t[j]
    })
    
    dummy := ListNode{0, nil}
    tail := &dummy
    
    for _, val := range t {
        tail.Next = &ListNode{val, nil}
        tail = tail.Next
    }
    
    return dummy.Next
}

File: Python/Data Structures/Linked List/Merge-k-Sorted-Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        nums = []
        
        for i in lists:
            current = i
            while current:
                nums.append(current.val)
                current = current.next
        
        nums.sort()
        
        dummy = ListNode(0)
        tail = dummy
        
        for i in nums:
            tail.next = ListNode(i)
            tail = tail.next
        
        return dummy.next
